gitpractice/javascript/task9.js

```diff
@@ -0,0 +1,48 @@
+function prefill(n, v) {
+    var array = [];
+    if (n === 0 || n==="0")
+        return [];
+    else if (typeof n == "string") {
+        if (isNaN(parseInt(n)))
+            throw new TypeError(n + " is invalid");
+        if (isNaN(n))
+            throw new TypeError(n + " is invalid");
+        //   for(var a=0;a<parseInt(n);a++)
+        //   array.push(v);
+        array = new Array(n);
+        array.fill(v, 0, n);
+    }
+    else if (typeof n == "number") {
+        if (n < 0)
+            throw new TypeError(n + " is invalid");
+        if (isNaN(n))
+            throw new TypeError(n + " is invalid");
+        if (n === Number.NEGATIVE_INFINITY || n === Number.POSITIVE_INFINITY)
+            throw new TypeError(n + " is invalid");
+        if (!Number.isInteger(n))
+            throw new TypeError(n + " is invalid");
```

Comment:
you can merge this if statements into one if statement, since you are throwing the same error for all the if statements